.github/scripts/generate_ci_workflows.py

```diff
@@ -9,77 +10,63 @@
 GITHUB_DIR = Path(__file__).parent.parent
 
 
-class PyTorchWindowsWorkflow:
+# it would be nice to statically specify that build_environment must be
+# present, but currently Python has no easy way to do that
+# https://github.com/python/mypy/issues/4617
+PyTorchWorkflow = Dict[str, Any]
+
+
+def PyTorchWindowsWorkflow(
+    *,
+    build_environment: str,
+    on_pull_request: bool = False
+) -> PyTorchWorkflow:
     CPU_TEST_RUNNER = "windows.4xlarge"
     CUDA_TEST_RUNNER = "windows.8xlarge.nvidia.gpu"
+    return {
+        "build_environment": build_environment,
+        "test_runner_type": (
+            CUDA_TEST_RUNNER
+            if "cuda" in build_environment
+            else CPU_TEST_RUNNER
```

Comment:
nit; Maybe it'd be better to make test_runner_type something that we explicitly pass in so that it's more apparent which workflows get which test runners.

Should make it much more apparent which workflows get which thing and reduces the amount of "magic" strings that we have

Reply:
sure, that's fine with me :+1: I just left it as-is in this PR because I was guessing you had a good reason for the way it already is, but I can change it before merging if you'd like